added conda-build3

matt-chan opened the pull request.

tovrstra commented on .travis.yml:
```diff
@@ -220,4 +223,4 @@ deploy:
     repo: ${GITHUB_REPO_NAME}
     tags: true
-    condition: "$TRAVIS_TAG == *a*"
+    condition: "$TRAVIS_TAG != *[ab]*"
```
Same comment as made elsewhere: `-l` should be added in front of `${ANACONDA_LABEL_ARG} and `condition` should be removed.

matt-chan replied:
I finally got it fixed... The sed line for ${ANACONDA_LABEL_ARG} needed a small tweak too.

matt-chan commented:
@tovrstra Ready for review. I think this should work well now. Sorry for all the version churn...

We should add Py3 to this, but we can do that after this is merged in. Also, I'm not a huge fan of all the different build paths (tagged, not tagged, PR). There's a lot to debug because of it. Also, it means that a PR that passes might not build on release. But maybe we should keep it for now because we're probably using all those build paths in the real development (we won't always build with conda-build I think?). We'll have to remember to tag PR builds to test conda builds. 

Also, I changed the build tags to be consistent with conda's documentation (dev/test/main instead of alpha/beta/main).

I also updated the template.py script to include an option for templating from a local file. That way we can apply travis scripts that haven't been merged to master yet (like this one). 

Things that we should still do, but probably in another PR/later:
- Py3
- Remove spurious requirements install in conda builds
- Add output for dynamic linking (ldd/otool)

tovrstra commented on .travis.yml:
```diff
@@ -116,6 +115,9 @@ install:
 - conda install -q python=${MYCONDAPY} cppcheck doxygen anaconda-client ${CONDA_DEPENDENCIES}
+# Deactivate build env and reactivate it to refresh compilers.
+- source deactivate && source $HOME/miniconda/bin/activate
+- hash -r
```
Would hash -r alone also work? Aslo, the activate script includes `hash -r`, so I guess it is not needed separately (if you decide to keep the deactivate/activate step).

matt-chan replied:
Ah I didn't know that about activate. No, I don't think hash itself works. There's some environmental variables that get set when the activate script is run again and which aren't run the first time. Since we install compilers, the paths are changing.

tovrstra replied:
OK. I think you can then remove the `hash -r` because it is already done in of the `miniconda/bin/activate` script.

tovrstra replied:
oh, sorry. it is already done. I was looking at the wrong diff. please ignore previous comment.

tovrstra commented on qcgrids/tests/CMakeLists.txt:
```diff
@@ -18,58 +18,50 @@
+                  DEPENDS check
+                  WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+                  COMMENT "Coverage analysis")
 
```
You can get rid of the targets refering to `tools/coverage_clean.sh` and `tools/coverage_analyze.py`. They were only used before we started working with CodeCov but the scripts no longer exist in the current repo.

matt-chan replied:
Okay!

tovrstra commented on qcgrids/tests/CMakeLists.txt:
```diff
@@ -18,58 +18,50 @@
-                          COMMENT "Running tests")
+add_custom_target(check
+                  COMMAND test_qcgrids
+                  DEPENDS test_qcgrids clean_coverage
```
The `clean_coverage` dependency can be removed and I guess then the if statement is no longer needed. the `check` target is just the same in both cases.

tovrstra commented:
@matt-chan This is already a great improvement! Really, thanks for the nice work! I only have a few minor comments and I agree the remaining points are better handled in separate PRs.

tovrstra commented:
@matt-chan Looks good. I'll check tomorrow that tests passed and merge but feel free to merge before I get to it.
